middle_end/flambda2/from_lambda/lambda_to_flambda_primitives.ml

```diff
@@ -836,78 +848,110 @@ let bigarray_set ~dbg ~unsafe kind layout b indexes value =
 
 (* Array accesses *)
 let array_access_validity_condition array array_kind index
-    ~(index_kind : L.array_index_kind) ~size_int =
+    ~(index_kind : L.array_index_kind) ~width_in_scalars ~size_int =
   [ multiple_word_array_access_validity_condition array ~size_int array_kind
-      index_kind ~width_in_scalars:1 ~index ]
+      index_kind ~width_in_scalars ~index ]
 
-let check_array_access ~dbg ~array array_kind ~index ~index_kind ~size_int
-    primitive : H.expr_primitive =
+let check_array_access ~dbg ~array (array_kind : P.Array_kind_for_length.t)
+    ~index ~index_kind ~size_int primitive : H.expr_primitive =
   checked_access ~primitive
     ~conditions:
       (array_access_validity_condition array array_kind index ~index_kind
-         ~size_int)
+         ~width_in_scalars:1 ~size_int)
     ~dbg
 
-let array_load_unsafe ~array ~index ~(mut : Lambda.mutable_flag)
-    (array_ref_kind : Array_ref_kind.t) ~current_region : H.expr_primitive =
+let array_load_unsafe ~array ~index ~(mut : Lambda.mutable_flag) array_kind
+    (array_ref_kind : Array_ref_kind.t) ~current_region : H.expr_primitive list
```

Comment:
Why do we take both `array_kind` and `array_ref_kind` here? Couldn't we use `convert_array_ref_kind_to_array_kind`?

Reply:
I've left a CR about this, will check once the remaining patches are merged.